src/main/kotlin/com/amazon/ionelement/impl/IonElementLoaderImpl.kt

```diff
@@ -89,18 +89,13 @@ internal class IonElementLoaderImpl(private val options: IonElementLoaderOptions
         return handleReaderException(ionReader) {
             val valueType = requireNotNull(ionReader.type) { "The IonReader was not positioned at an element." }
 
-            val annotations = ionReader.typeAnnotations!!.asList().toEmptyOrPersistentList()
+            val annotations = ionReader.typeAnnotations!!.toImmutableListUnsafe()
```

Comment:
🗺️ Previously, this line was creating a list from an array, and then creating a persistent list by copying all of the elements from the intermediate list (an `O(n)` operation). Now, it just wraps the array with an `ArrayBackedImmutableList` (an `O(1)` operation) in addition to `ArrayBackedImmutableList` having a smaller object layout to begin with.